### ISSUE TITLE
Web UI not seeing installing apps after update

### ISSUE THREAD

**Issue by randomsock** [tableflip/guvnor#60]
After `guv kill`, `npm update`, then restarting guv, the web UI says "No apps are installed", but terminal `guv lsapps` shows apps still installed (as expected).  Not sure whether this was as a result of the update or the restart, but gut feeling is the restart.

**Comment by TomKaltz** [comment]
I'm wondering if it's still standard operating procedure to do the following to upgrade...
```
guv kill
sudo npm remove -g guvnor
sudo npm install -g guvor
```
That may be your problem.

**Comment by randomsock** [comment]
Yeah, must admit I'd got lazy about that.  Just tried the full thing again but still no joy.

Thanks for the suggestion though.

**Comment by achingbrain** [comment]
@TomKaltz It's still standard procedure, yes.  This is because npm has been inconsistent with how `update` behaves over versions and sometimes will update the top level module but not any dependencies which is not really what you want.  It's more reliable to just blow the old version away and then reinstall.  Also `--unsafe-perm` should be used when installing otherwise you get lots of warnings and the occasional failure with native modules not being able to access /root/.node-gyp when compiling.

@randomsock what is your setup like?  Are you running guvnor-web under the instance of guvnor that you are upgrading or standalone/on another machine?  Does reloading the page after restarting guvnor make any difference?

**Comment by randomsock** [comment]
At the moment, yes I'm running guv and web on the same instance.  It's my own dev server so I'll trash it and have a play with some sequences, see if I can find a pattern.

**Comment by randomsock** [comment]
It's killing guvnor itself that does it.

Restarting guv-web doesn't affect it - apps still show in the UI.

`guv kill` then `guv` to restart (no update or anything between) shows guv-web and apps restarting, then when the UI reloads, it says "No apps are installed".

`guv lsapps` lists the apps ok, and they are still in `/etc/guvnor/apps.json`.

Closing and reloading the browser doesn't bring it back.

Hope that helps.
